wallet-new/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -822,6 +845,72 @@ instance ToSchema TransactionDirection where
             & type_ .~ SwaggerString
             & enum_ ?~ ["outgoing", "incoming"]
 
+-- | This is an information-less variant of 'PtxCondition'.
+data TransactionStatus
+    = Applying
+    | InNewestBlocks
+    | Persisted
+    | WontApply
+    | Creating
+    deriving (Eq, Ord, Show, Enum, Bounded)
+
+transactionStatusToText :: TransactionStatus -> Text
+transactionStatusToText x = case x of
+    Applying {} ->
+        "applying"
+    InNewestBlocks {} ->
+        "inNewestBlocks"
+    Persisted {} ->
+        "persisted"
+    WontApply {} ->
+        "wontApply"
+    Creating {} ->
+        "creating"
+
+instance ToJSON TransactionStatus where
+    toJSON x = object
+        [ "tag" .= transactionStatusToText x
+        , "data" .= Object mempty
+        ]
+
+instance ToSchema TransactionStatus where
+    declareNamedSchema _ =
+        pure $ NamedSchema (Just "TransactionStatus") $ mempty
+            & type_ .~ SwaggerObject
+            & required .~ ["tag", "data"]
+            & properties .~ (mempty
+                & at "tag" ?~ Inline (mempty
+                    & type_ .~ SwaggerString
+                    & enum_ ?~
+                        [ "applying", "inNewestBlocks", "persisted"
+                        , "wontApply", "creating"
+                        ]
+                )
+                & at "data" ?~ Inline (mempty
+                    & type_ .~ SwaggerObject
+                )
+            )
+
+instance FromJSON TransactionStatus where
+    parseJSON = withObject "TransactionStatus" $ \o -> do
+       tag <- o .: "tag"
+       case tag of
+           "applying" ->
+                pure Applying
+           "inNewestBlocks" ->
+                pure InNewestBlocks
+           "persisted" ->
+                pure Persisted
+           "wontApply" ->
+                pure WontApply
+           "creating" ->
```

Comment:
Mental note: we should have some sort of manifesto/guidelines to be sure we are consistent in how we are rendering enumerations in the API, as there is always the choice between snake_case, camelCase etc. Let's pick one style and be consistent 😁

Reply:
I have only seen camel case thus far